VAS.Core/Common/StateController.cs

```
@@ -57,33 +57,38 @@ public StateController ()
 			Log.Debug ("Moving to " + transition);
 
 			if (!destination.ContainsKey (transition)) {
-				Log.Debug ("Moving failed because transition" + transition + " is not in destination dictionary.");
+				Log.Debug ("Moving failed because transition " + transition + " is not in destination dictionary.");
 
 				return false;
 			}
 
-			bool isModal = false;
-			IScreenState lastState = LastState (out isModal);
-			if (lastState != null) {
-				bool postTransition = await lastState.PostTransition ();
-				if (!postTransition) {
-					Log.Debug ("Moving failed because panel " + lastState.Panel.PanelName + " cannot move.");
-					return false;
+			try {
+				bool isModal = false;
+				IScreenState lastState = LastState (out isModal);
+				if (lastState != null) {
+					bool postTransition = await lastState.PostTransition ();
+					if (!postTransition) {
+						Log.Debug ("Moving failed because panel " + lastState.Panel.PanelName + " cannot move.");
+						return false;
+					}
 				}
-			}
 
-			if (isModal) {
-				await PopAllModalStates ();
-			}
+				if (isModal) {
+					await PopAllModalStates ();
+				}
 
-			IScreenState state = destination [transition] ();
-			bool ok = await state.PreTransition (data);
-			if (ok) {
-				await PushNavigationState (transition, state);
-			} else {
-				Log.Debug ("Moving failed because panel " + state.Panel.PanelName + " cannot move.");
+				IScreenState state = destination [transition] ();
+				bool ok = await state.PreTransition (data);
+				if (ok) {
+					await PushNavigationState (transition, state);
+				} else {
+					Log.Debug ("Moving failed because panel " + state.Panel.PanelName + " cannot move.");
+				}
+				return ok;
+			} catch (Exception ex) {
```

You don't really know what can fail in a transition. This try/catch is merely for debugging purpose since exceptions were simply ignored and not even printed. If we need it, we can even display a warning to the user, instead of just bailing out